### PR TITLE
[Backport 8.1]: Add new API to schedule async Task on ScheduleService

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description
Backports https://github.com/camunda/zeebe/pull/11788

## Related issues

related to https://github.com/camunda/zeebe/issues/11761

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: engine/src/main/java/io/camunda/zeebe/streamprocessor/StreamProcessor.java [hunk 1]
```diff
@@ -274,6 +288,24 @@ private void healthCheckTick() {
     actor.runDelayed(HEALTH_CHECK_TICK_DURATION, this::healthCheckTick);
   }
 
+  void chainSteps(final int index, final Step[] steps, final Runnable last) {
```
Comment (oleschoenburg) [inline engine/src/main/java/io/camunda/zeebe/streamprocessor/StreamProcessor.java:291]: :wrench: let's make this private at first

File: engine/src/main/java/io/camunda/zeebe/streamprocessor/StreamProcessor.java [hunk 2]
```diff
@@ -538,6 +568,36 @@ public void onRecordAvailable() {
...
+            }
+          });
+    }
+
+    public ActorControl getActorControl() {
+      return actor;
+    }
+  }
+
+  public interface Step {
```
Comment (oleschoenburg) [inline engine/src/main/java/io/camunda/zeebe/streamprocessor/StreamProcessor.java:597]: :wrench: Same here, should start off as private I think

**Review by lenaschoenburg — APPROVED** [review]

File: engine/src/main/java/io/camunda/zeebe/streamprocessor/StreamProcessor.java (hunk 1 above)
Comment (oleschoenburg) [reply engine/src/main/java/io/camunda/zeebe/streamprocessor/StreamProcessor.java:291]: We can also do this later to keep our backports and other PRs usable as is.

**Pull request merged**